pytest_xprocess.py

```diff
@@ -64,12 +55,40 @@ def pytest_runtest_makereport(item, call):
                 longrepr.addsection("%s log" % name, content)
 
 
+def pytest_unconfigure(config):
+    try:
+        config._xprocess._clean_up_resources()
+    except AttributeError:
+        # xprocess fixture was not used
+        pass
+
+
 def pytest_configure(config):
-    config.__dict__.setdefault("_xprocess_file_handles", [])
+    config.pluginmanager.register(InterruptionHandler())
 
 
-def pytest_unconfigure(config):
-    """All logfile handles should be closed by the end of the test run.
-    This is done in order to avoid ResourceWarnings."""
-    for f in config._xprocess_file_handles:
-        f.close()
+class InterruptionHandler:
+    """The purpose of this class is exposing the
+    config object containing references necessary
+    to properly clean-up in the event of an exception
+    during test runs"""
+
+    def pytest_configure(self, config):
+        self.config = config
+
+    def info_objects(self):
+        return self.config._xprocess._info_objects
+
+    def interruption_clean_up(self):
+        try:
+            for info in self.info_objects():
+                info.terminate()
+            self.config._xprocess._clean_up_resources()
+        except AttributeError:
```

Comment:
Same tip as above: avoid catching generic exceptions like `AttributeError` in broad try/except clauses as it might mask errors.

```python
    def interruption_clean_up(self):
        try:
            xprocess = self.config._xprocess
        except AttributeError:
            pass
        else:                
            for info in self.info_objects():
                info.terminate()
            xprocess._clean_up_resources()
```

Reply:
👍🏻